source/MaterialXCore/Types.cpp

```diff
@@ -185,10 +220,47 @@ template <> Matrix44 MatrixN<Matrix44, float, 4>::getAdjugate() const
         _arr[0][0]*_arr[2][1]*_arr[1][2] - _arr[1][0]*_arr[0][1]*_arr[2][2] - _arr[2][0]*_arr[1][1]*_arr[0][2]);
 }
 
+
+Vector4 Matrix44::transform(const Vector4& rhs) const
+{
+    return Vector4(
+      _arr[0][0] * rhs[0] + _arr[0][1] * rhs[1] + _arr[0][2] * rhs[2] + _arr[0][3] * rhs[3],
+      _arr[1][0] * rhs[0] + _arr[1][1] * rhs[1] + _arr[1][2] * rhs[2] + _arr[1][3] * rhs[3],
+      _arr[2][0] * rhs[0] + _arr[2][1] * rhs[1] + _arr[2][2] * rhs[2] + _arr[2][3] * rhs[3],
+      _arr[3][0] * rhs[0] + _arr[3][1] * rhs[1] + _arr[3][2] * rhs[2] + _arr[3][3] * rhs[3]
+    );
+}
+
+Vector3 Matrix44::transformPoint(const Vector3& rhs) const
+{
+    Vector4 rhsVector4(rhs[0], rhs[1], rhs[2], 1.0f);
+    rhsVector4 = transform(rhsVector4);
+    return Vector3(rhsVector4[0], rhsVector4[1], rhsVector4[2]);
+}
+
+Vector3 Matrix44::transformVector(const Vector3& rhs) const
+{
+    Vector4 rhsVector4(rhs[0], rhs[1], rhs[2], 0.0f);
+    rhsVector4 = transform(rhsVector4);
+    return Vector3(rhsVector4[0], rhsVector4[1], rhsVector4[2]);
+}
+
+Vector4 Matrix44::transformNormal(const Vector4& rhs) const
```

Comment:
What would be the proposed use case for this Vector4 overload of transformNormal?  Perhaps we can omit this and provide only the more common Vector3 version?

Reply:
It was convenient for something I did in Mesh.cpp with it. Plus, the MaterialX spec's transformnormal supports Vector4s so I was trying to be consistent.

Reply:
Got it, and thanks for those additional details.  This seems fine to me.

Reply:
Got it, and thanks for those additional details.  This seems fine to me.

Reply:
It looks like the vector4 version of transformnormal in the specification was an error, so I'm re-opening this thread.  Given that additional context, I would recommend that we remove this overload of transformNormal in the code, and replace it with the equivalent math in Mesh.cpp.